src/pybind/mgr/volumes/fs/operations/volume.py

```diff
@@ -227,7 +247,56 @@ def list_volumes(mgr):
         result.append({'name': f['mdsmap']['fs_name']})
     return result
 
-
+def info(mgr, path, fs_handle, volname):
+    """
+    Get all information of a CephFS volume.
+    """
+    vol_info_dict = {} # type: Dict[str, Any]
+    df = mgr.get("df")
+    pool_stats = dict([(p['id'], p['stats']) for p in df['pools']])
+    osdmap = mgr.get("osd_map")
+    pools = dict([(p['pool'], p) for p in osdmap['pools']])
+    metadata_pool_id, data_pool_ids = get_pool_ids(mgr, volname)
+    vol_info_dict["pools"] = {"metadata" : [], "data" : []}
+    for pool_id in [metadata_pool_id] + data_pool_ids:
+        if pool_id == metadata_pool_id:
+            pool_type = "metadata"
+        else:
+            pool_type = "data"
+        vol_info_dict["pools"][pool_type].append({
+                        'name' : pools[pool_id]['pool_name'],
+                        'used': pool_stats[pool_id]['bytes_used'],
+                        'avail': pool_stats[pool_id]['max_avail']})
+                             
+    mon_addr_lst = []
+    mon_map_mons = mgr.get('mon_map')['mons']
+    for mon in mon_map_mons:
+        ip_port = mon['addr'].split("/")[0]
+        mon_addr_lst.append(ip_port)
+    vol_info_dict["mon_addrs"] = mon_addr_lst
+    try:
+        st = fs_handle.statx(path.encode('utf-8'), cephfs.CEPH_STATX_SIZE,
+                                cephfs.AT_SYMLINK_NOFOLLOW)
+                                
+        usedbytes = st['size']
+        num_groups = len(listdir(fs_handle, path, filter_entries=[Trash.GROUP_NAME.encode('utf-8')]))
+        trashdir = os.path.join(path, Trash.GROUP_NAME)
+        try:
+            num_pending_subvol_del = len(os.listdir(trashdir))
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                num_pending_subvol_del = 0
+                
+        vol_info_dict.update({'volume_name' : volname,
+                                'size' : usedbytes,
+                                'subvolumegroups' : num_groups,
+                                'pending_subvolume_deletions' : num_pending_subvol_del})
+    except cephfs.Error as e:
+        if e.args[0] == errno.ENOENT:
```

Comment:
What's the reason to handle ENOENT here?

Reply:
https://github.com/ceph/ceph/pull/45949#discussion_r858597314

Reply:
OK -- in that case, isn't it better to stat(path) in caller and exit early on? That way, this routine can become simpler.

Reply:
You mean to say if `/volumes` doesn't exit, there's no need to display other info like `mon_addrs,pools,volume_name` .Cause, currently in this case it will show these details only leaving the details related to subvolumegroups etc .

Reply:
Currently, it is like first collect information that not related with `/volumes` directory and then check if path exists extend these info with others else return with those info only.

Reply:
Right - the helper need not worry about /volumes not being available.